### PR TITLE
check mapping interface in dict trafaret

### PR DESCRIPTION
**Pull request by jettify**
For now `Dict` trafaret checks that input value is instance of  `dict`, but it is very useful to relax this constraint and check input only against `abc.Mapping`.  So any object with dict like interface could be validated.

### CODE REVIEW

File: trafaret/__init__.py [hunk 1]
```diff
@@ -999,9 +999,9 @@ def make_optional(self, *args):
         return self
 
     def check_and_return(self, value):
-        if not isinstance(value, dict):
-            self._failure("value is not dict")
-        data = copy.copy(value)
+        if not isinstance(value, AbcMapping):
+            self._failure("value is not dict like object")
```
Comment (asvetlov) [inline trafaret/__init__.py]: dict-like is for duck-typing. Don't hesitate to use `Mapping` word.

**Comment by jettify** [comment]
I've changed error message to: "value does not provide Mapping interface"

**Comment by asvetlov** [comment]
Cool! 
LGTM!

**Comment by asvetlov** [comment]
Hmm. After second look I'm revoking my agreement.
I guess your change should support miltidicts also (I mean dictionary with duplicate keys).

File: trafaret/__init__.py [hunk 2]
```diff
@@ -999,9 +999,9 @@ def make_optional(self, *args):
         return self
 
     def check_and_return(self, value):
-        if not isinstance(value, dict):
-            self._failure("value is not dict")
-        data = copy.copy(value)
+        if not isinstance(value, AbcMapping):
+            self._failure("value does not provide Mapping interface")
+        data = {k: v for k, v in value.items()}
```
Comment (asvetlov) [inline trafaret/__init__.py:1004]: The line reduces multidict dropping duplicate keys.

**Comment by Deepwalker** [comment]
Implemented change in other way. Reused some code.

**Pull request closed without merging**